devito/finite_differences/differentiable.py

```diff
@@ -145,24 +186,48 @@ def __eq__(self, other):
         return super(Differentiable, self).__eq__(other) and\
             all(getattr(self, i, None) == getattr(other, i, None) for i in self._state)
 
+    def index(self, dim):
+        for d in self.dimensions:
+            if d is dim:
+                return d
+        return None
+
+    @property
+    def name(self):
+        return "".join(f.name for f in self._functions)
+
     @property
     def laplace(self):
         """
         Generates a symbolic expression for the Laplacian, the second
         derivative w.r.t all spatial Dimensions.
         """
-        space_dims = [d for d in self.indices if d.is_Space]
+        space_dims = [d for d in self.dimensions if d.is_Space]
```

Comment:
Can this be made a property as is done with `Function`'s?

Reply:
there's already `space_dimensions` I think

Reply:
ah yes, only in Function, dumb me. We could **move** it from Function to Differentiable though